app/javascript/react/components/Map/Markers/CrowdMapMarkers.tsx

```diff
@@ -90,21 +104,81 @@ const CrowdMapMarkers = ({ pulsatingSessionId, sessions }: Props) => {
     }
   }, [dispatch, rectanglesRef.current.length, crowdMapRectanglesLength]);
 
+  useEffect(() => {
+    const handleMapInteraction = () => {
+      dispatch(clearRectangles());
+    };
+
+    map && map.addListener("click", handleMapInteraction);
+    map && map.addListener("touchend", handleMapInteraction);
+    map && map.addListener("dragstart", handleMapInteraction);
+    map && map.addListener("zoom_changed", handleMapInteraction);
+
+    return () => {
+      if (map) {
+        google.maps.event.clearListeners(map, "click");
+        google.maps.event.clearListeners(map, "touchend");
+        google.maps.event.clearListeners(map, "dragstart");
+        google.maps.event.clearListeners(map, "zoom_changed");
+      }
+    };
```

Comment:
just a food for thought - we use almost the same logic in fixed markers - it would be good to extract it into a separate utility function which would take on event names as argument  🤔